### PR TITLE
[NF-8882] Add Validation to the Decisions API for Local Status

### PR DESCRIPTION
**Pull request by zefta**
Please note that this PR comes with a PR on WA.

### CODE REVIEW

File: designation_errors.md [hunk 1]
```diff
@@ -17,3 +17,19 @@ HTTP/1.1 422 Unprocessable Entity
...
+{
+  "errors": {
+    "decision_id": [
+      "Cannot set the decision_id when the local_status_id is not null"
+    ]
+  }
+}
+```
+
+You'll get this error when you're trying to set a decision_id to a designation which already has a local status assigned. If you want to be able to force the decision_id please contact Liaison to turn on the "Disassociate Decision Codes from Local Status" feature which will allow you to set the decision even if the local_status is present.
```
Comment (benjaminoakes) [inline designation_errors.md]: We should maybe use backticks around `decision_id` and `local_status`.

File: designation_errors.md [hunk 2]
```diff
@@ -17,3 +17,19 @@ HTTP/1.1 422 Unprocessable Entity
...
+
+```
+HTTP/1.1 422 Unprocessable Entity
+```
+
+```json
+{
+  "errors": {
+    "decision_id": [
+      "Cannot set the decision_id when the local_status_id is not null"
```
Comment (benjaminoakes) [inline designation_errors.md]: Should this be `local_status_id` or `local_status`?  (The API doesn't expose anything called `local_status_id`.)

Comment (benjaminoakes) [inline designation_errors.md]: We say `"can't be blank"` so shouldn't this be `"can't set ..."` to match?

**Comment by zefta** [comment]
@benjaminoakes comments addressed

**Comment by zefta** [comment]
@Dana-Maxfield this PR should have been merged already, the ticket is marked as "Resolved"

**Comment by Dana-Maxfield** [comment]
@Zefta Same here, I only merge those that are marked as ready for RM (or the equivalent).

**Pull request merged**